bpf-gpl/parsing.h

```diff
@@ -36,19 +49,21 @@ static CALI_BPF_INLINE int parse_packet_ip(struct cali_tc_ctx *ctx) {
 		}
 	default:
 		if (CALI_F_WEP) {
-			CALI_DEBUG("Unknown ethertype (%x), drop\n", bpf_ntohs(ctx->skb->protocol));
+			CALI_DEBUG("Unknown ethertype (%x), drop\n", protocol);
 			goto deny;
 		} else {
 			CALI_DEBUG("Unknown ethertype on host interface (%x), allow\n",
-								bpf_ntohs(ctx->skb->protocol));
+								protocol);
 			goto allow_no_fib;
 		}
 	}
 
-	if (skb_refresh_validate_ptrs(ctx, UDP_SIZE)) {
-		ctx->fwd.reason = CALI_REASON_SHORT;
-		CALI_DEBUG("Too short\n");
-		goto deny;
+	if (!CALI_F_XDP) {
+		if (skb_refresh_validate_ptrs(ctx, UDP_SIZE)) {
+			ctx->fwd.reason = CALI_REASON_SHORT;
+			CALI_DEBUG("Too short\n");
+			goto deny;
+		}
```

Comment:
This branch is similar to the one at line 24, and can be merged together. The downside is that for we will use more cpu cycle on some switch cases. What do you think?

Reply:
I think it's best to leave like this, to save those CPU cycles in the TC case.